framework/auth/core.py

```diff
@@ -164,10 +165,18 @@ def logged_in(self):
     def from_kwargs(cls, request_args, kwargs):
         user = request_args.get('user') or kwargs.get('user') or _get_current_user()
         private_key = request_args.get('view_only')
-
+        try:
+            # Avoid circular import
+            from website.project.model import PrivateLink
+            private_link = PrivateLink.find_one(
+                Q('key', 'eq', private_key)
```

Comment:
Furthermore, when `private_key` is `None`, this will match *all* private links. When there is more than one result, `find_one` will raise `MultipleResultsFound`, which is a subclass of `QueryException` and will therefore land in the `private_link = None` codepath just below. However, if ever there is exactly one private link, this code will match it no matter what: that's a bug, though admittedly highly unlikely to occur in production. It's easy to fix though: let's make the private_link query conditional on `private_key is not None`. Note that this is preferable to switching default for `private_key` to `''`, since there could theoretically be a private link where `key == ''`.